plugin.cpp

```diff
@@ -36,35 +36,54 @@ static const char *default_config = QUOTE({
 			"type": "boolean",
 			"displayName": "Enabled",
 			"default": "false",
-			"order" : "5"
+			"order" : "7"
 		       	},
+        "toleranceMeasure": {
+			"description": "Whether tolerance is specified as a percentage or in absolute terms",
+			"type": "enumeration",
+			"options" : [ "percentage", "absolute value" ],
+			"default": "percentage",
+			"order" : "1",
+			"displayName" : "Tolerance Measure"
+			},
 		"tolerance": {
-			"description": "A percentage difference that will be tolerated when determining if values are equal.",
+			"description": "A percentage/absolute difference that will be tolerated when determining if values are equal.",
 			"type": "float",
-			"default": "0",
-			"order" : "1",
-		       	"displayName" : "Tolerance %"
+			"minimum": "0.0",
+			"default": "1.0",
+			"order" : "2",
+			"displayName" : "Tolerance Value"
+			},
+		"processingMode": {
+			"description": "Reading processing mode",
+			"type": "enumeration",
+			"options" : [ "Include full reading if any Datapoint exceeds tolerance", "Include full reading if all Datapoints exceed tolerance", 
+                            "Include only the Datapoints that exceed tolerance" ],
+			"default": "Include full reading if any Datapoint exceeds tolerance",
+			"order" : "3",
+			"displayName" : "Reading Processing Mode"
 			},
 		"minRate": {
 			"description": "The minimum rate at which data must be sent",
 			"type": "integer",
+			"minimum": "0",
 			"default": "0",
-			"order" : "2",
+			"order" : "4",
 			"displayName" : "Minimum Rate"
 			},
 		"rateUnit": {
 			"description": "The unit used to evaluate the minimum rate",
 			"type": "enumeration",
 			"options" : [ "per second", "per minute", "per hour", "per day" ],
 			"default": "per second",
-			"order" : "3",
+			"order" : "5",
 			"displayName" : "Minimum Rate Units"
 			},
 		"overrides" : {
 			"description": "Individual asset tolarances if different from the global tolarance",
 			"type": "JSON",
 			"default": "{ }",
-			"order" : "4",
+			"order" : "6",
 			"displayName" : "Individual Tolerances"
```

Comment:
We better add `"group": "Individual Tolerances"` as well.

Reply:
If the scope of the setting for tolerances includes the individual tolerances then we should keep all the tolerances together in one group call "Tolerances".

The processing mode (which is really output mode), rate and rate unit can go in a group call "Output".

Isolating individual tolerances like this is probably a mistake, although if we switch to using a list type for these then we are forced to by the current GUI implementation.

Reply:
Yes please. Remove that group thing.